libminifi/src/io/tls/TLSSocket.cpp

```diff
@@ -200,17 +208,12 @@ int16_t TLSSocket::initialize(bool blocking) {
         logger_->log_trace("want read");
         return 0;
       } else {
+        logger_->log_error("SSL socket connect failed to %s %d", requested_hostname_, port_);
+        SSL_free(ssl_);
+        ssl_ = NULL;
+        Socket::closeStream();
```

Comment:
These lines for cleanup are present in multiple places in the file. Did you consider extracting to a function or utilizing RAII for cleanup?

Reply:
Dtor cleans up anyway, but I was not brave enough to assume that socket objects are never reused. :) 

Simplified cleanup logic anyway.